### PR TITLE
Remove "is final document state" from compilation trackers.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Idea made by @jasonmalinowski  in prior prs.  Followup to https://github.com/dotnet/roslyn/pull/72021

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski this is ready for review.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski @ToddGrun ptal.

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski @ToddGrun ptal.

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Comment by CyrusNajmabadi** [comment]
Merging in.  But would still like a review from @jasonmalinowski .

File: src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs [hunk 1]
```diff
@@ -732,14 +731,15 @@ async Task<InProgressState> CollapseInProgressStateAsync(InProgressState initial
...
                         // We will finalize the compilation by adding full contents here.
                         var (compilationWithGeneratedDocuments, generatedDocuments, generatorDriver) = await AddExistingOrComputeNewGeneratorInfoAsync(
+                            isFrozen,
                             compilationState,
                             compilationWithoutGeneratedDocuments,
                             generatorInfo,
                             staleCompilationWithGeneratedDocuments,
                             cancellationToken).ConfigureAwait(false);
 
                         // After producing the sg documents, we must always be in the final state for the generator data.
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs:741]: Comment is stale (kinda).

**Comment by jasonmalinowski** [comment]
This looked good @CyrusNajmabadi.

**Pull request merged**